Destination/StreamDestination.php

```diff
@@ -23,6 +32,53 @@ public function push($filename, LoggerInterface $logger)
     {
         $logger->info(sprintf('Copying %s to %s', $filename, $this->directory));
 
-        copy($filename, sprintf('%s/%s', $this->directory, basename($filename)));
+        copy($filename, ($destination = sprintf('%s/%s', $this->directory, basename($filename))));
+        touch($destination, filemtime($filename));
+
+        if (is_array($this->backups)) {
+            $this->backups[] = Backup::fromFile($destination);
+        } else {
+            $this->lazyLoadBackups();
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getIterator()
+    {
+        if (!is_array($this->backups)) {
+            $this->lazyLoadBackups();
+        }
+
+        return new ArrayIterator($this->backups);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function count()
+    {
+        if (!is_array($this->backups)) {
+            $this->lazyLoadBackups();
+        }
+
+        return count($this->backups);
+    }
+
+    /**
+     * Loads list of backups in lazy manner.
+     */
+    protected function lazyLoadBackups()
```

Comment:
Agree.

> On Sep 11, 2015, at 3:28 PM, Kevin Bond <notifications@github.com> wrote:
> 
> In Destination/StreamDestination.php <https://github.com/kbond/ZenstruckBackupBundle/pull/3#discussion_r39270478>:
> 
> > +    /**
> > +     * {@inheritdoc}
> > +     */
> > +    public function count()
> > +    {
> > +        if (!is_array($this->backups)) {
> > +            $this->lazyLoadBackups();
> > +        }
> > +
> > +        return count($this->backups);
> > +    }
> > +
> > +    /**
> > +     * Loads list of backups in lazy manner.
> > +     */
> > +    protected function lazyLoadBackups()
> Instead of having checks for is_array checks all over the place, just have:
> 
> protected function getBackups()
> {
>     if (null !== $this->backups) {
>         return $this->backups;
>     }
> 
>     // ... contents of lazyloadbackups
> }
> That way you can just call $this->getBackups() any time you need them.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/kbond/ZenstruckBackupBundle/pull/3/files#r39270478>.
>